source/DSCResources/DSC_CMForestDiscovery/DSC_CMForestDiscovery.psm1

```diff
@@ -0,0 +1,299 @@
+
+                foreach ($item in $array)
+                {
+                    if (($desiredPollingSchedule).$($item) -ne ($stateSchedule).$($item))
+                    {
+                        Write-Verbose -Message ($script:localizedData.ScheduleItem `
+                            -f $item, $($desiredPollingSchedule.$($item)), $(($stateSchedule).$($item)))
+                        $setSchedule = $true
+                    }
+                }
+
+                if ($setSchedule)
+                {
+                    $buildingParams += @{
+                        PollingSchedule = $desiredPollingSchedule
+                    }
+                }
+            }
+
+            if ($buildingParams)
+            {
+                Set-CMDiscoveryMethod -ActiveDirectoryForestDiscovery -SiteCode $SiteCode @buildingParams
+            }
+        }
+        else
+        {
+            if ($state.Enabled -eq $true)
+            {
+                Set-CMDiscoveryMethod -ActiveDirectoryForestDiscovery -SiteCode $SiteCode -Enabled $false
+            }
+        }
+    }
+    catch
+    {
+        throw $_
+    }
+    finally
+    {
+        Set-Location -Path "$env:temp"
+    }
+}
+<#
+    .SYNOPSIS
+        This will test the desired state.
+
+    .PARAMETER SiteCode
+        Specifies the site code for Configuration Manager site.
+
+    .PARAMETER Enabled
+        Specifies the enablement of the forest discovery method. If settings is set to $false no other value provided will be
+        evaluated for compliance.
+
+    .PARAMETER PollingSchedule
+        Specifies a schedule and determines how often Configuration Manager attempts to discover Active Directory forest.
+
+    .PARAMETER EnableActiveDirectorySiteBoundaryCreation
+        Indicates whether Configuration Manager creates Active Directory boundaries from AD DS discovery information.
+
+    .PARAMETER EnableSubnetBoundaryCreation
+        Indicates whether Configuration Manager creates IP address range boundaries from AD DS discovery information.
+#>
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $Enabled,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance]
+        $PollingSchedule,
+
+        [Parameter()]
+        [Boolean]
+        $EnableActiveDirectorySiteBoundaryCreation,
+
+        [Parameter()]
+        [Boolean]
+        $EnableSubnetBoundaryCreation
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -Enabled $Enabled
+    $result = $true
+
+    if ($Enabled -eq $true)
+    {
+        $exludelist = @('Verbose','PollingSchedule')
```

Comment:
Same as above I would change this to an include list.

Reply:
Swapped out for an includeList.